### DISCUSSION TITLE
Quero Aprender JavaScript. Mas, a Faculdade Quer Me Ensinar JAVA

### DISCUSSION THREAD

**Discussion by henriquebrg43** [frontendbr/forum#922] [Discussão]
Olá, meu nome é Henrique. Tenho 24 anos e me encontro na seguinte situação:

Gosto da "parte" de **Front End** e tenho a vontade de seguir esse caminho (já que, fiz alguns cursos: _web designer, Programador Web_. Assisto Palestras, Acompanho o Fórum, Slack, estou sempre me atualizando sobre a área).

Ainda só sei o básico de **HTML e CSS** (Até consigo montar sites. Mas, não são responsivos)

Minha intenção esse ano é (ou era) me aprofundar em **HTML, CSS & JavaScript**

Mas, neste ano, inicío o 2° ano do curso **Análise e Desenvolvimento de Sistemas** (são 3 anos),

E apartir de agora até o último ano "só" veremos **JAVA**, que é uma linguagem que não me desperta interesse, mesmo pesquisando sobre e assistindo palestras. Ao contrário de JavaScript (que me deixa muito interessado). Já que não tenho muito conhecimento em programação, eu teria que utilizar grande parte do meu tempo para entender/estudar **JAVA**. Sendo assim, me deixaria com pouquissimo (ou até nenhum) tempo  para estudar "**Front End**".

Possivelmente, alguem aqui já passou por algum dilema (parecido ou não).

Qual foi seu dilema? e porque escolheu **A** ou **B**? (*Se quiser me deixar uma dica de como me portar nessa situação, eu serei grato)

Talvez eu não esteja dando/vendo o devido valor do **java** no **Front End** (se você puder/quiser, mostre-me)

*detalhe:
estudo em um Intituto Federal, minha cidade (interior) tem menos de 100k de habitantes, todos os cursos superios que tem são voltados para **java**, as empresas da cidade "respiram" **java**, exceto algumas pequenas agências, (abandonar o curso não é uma opção hoje).

**Comment by luizbills** [comment]
Na minha época e possívelmente a de muitos outros aprediamos **linguagem C** (que é muito mais chato que Java). 
Enfim, estude o Java pois você precisa de passar nessa diciplina. E um outro motivo é que Java vai te ajudar a começar a ter pensamento lógico (que é a base da programação) já que você ainda está engatinhando nessa área.

> Talvez eu não esteja dando/vendo o devido valor do java no Front End 

e nunca vai ver kkkkkkkk java é back-end

**Reply by Vanessandrade7** [reply]
Estude o que vc quer, tem pessoas metódicas que acham que só tem sucesso quem faz universidade, se vc for bom no que faz vc vai longe com ou sem Java

**Comment by tcelestino** [comment]
Você não precisa da faculdade para aprender Javascript. Até porque a faculdade só vai te apresentar a teoria e a prática por baixo. Ou seja, comece a estudar o que você curte por fora. Existem diversas fontes de conteúdo pela a internet a fora.

Outra coisa, a questão do Java na faculdade é apenas para você entender/aprender lógica de programação,  o conceito de orientado objeto e que no final você pode 'levar' para qualquer linguagem que queira usar no futuro. Nunca vi nenhuma faculdade ensinar lógica de programação com JS e na minha opinião nem faz sentido. Logo, por mais que você não curta, tente tirar o máximo que no final vai precisar no futuro.

**Comment by henriquebrg43** [comment]
@luizbills @tcelestino Gratidão pelos comentários (ajudou bastante).

**Comment by kivervinicius** [comment]
@henriquebrg43 Sou fullstack hoje, e te digo uma coisa, grandes coisas que sei hoje foi devido ao incentivo na faculdade, minha dica é, cai de cabeça no curso, finaliza o mais rápido possível, se dedica a isso, o front-end é minha paixão também, mas até eu conseguir trabalhar na área passei por inúmeras linguagens back-end, já que sempre essa foi a demanda na área (sou da época de 2006), Fui professor em uma faculdade da minha cidade e sei o que está passando pois muitos alunos reclamavam da mesma coisa, você não precisa "respirar" Java, mas te garanto que se você for mediano no Java e muito bom no Front la na frente isso vai te valer muito!

**Comment by JosielLima** [comment]
Aproveita a graduação e tente extrair o máximo de conceitos de programação, isso irá te ajudar em qualquer linguagem no futuro.

Sobre Front-end estude por fora. É uma área que possui muito conteúdo na Web, recomendo essa excelente plataforma gratuita:  https://www.freecodecamp.org/

**Comment by AmandaFonseca** [comment]
Eu fiz analise de sistemas também mas não foi numa instituição publica geralmente num curso do tipo tecnologo o tempo é curto e quantidade de matérias são muitas então de qualquer maneira você vai ter que estudar depois. Acredito que a faculdade seja apenas uma apresentação de conceitos e carreiras que o aluno pode seguir, muitas pessoas estudam analise de sistemas e depois migram para segurança da informação por exemplo. No caso estudar JAVA acaba sendo extremamente útil pois você irá desenvolver sua lógica, trabalhar com funções, loops e arrays ou seja, aprende conceitos que também estão presentes no JavaScript. Se dedique  a aprender Java afinal ele está na grade do curso e a faculdade não vai tirar e quando mais rápido você se livra de uma matéria mais de boa você fica para aprender o que você realmente gosta.

**Comment by silvaMatheus** [comment]
Eu pegaria os exercícios propostos em java e tentaria fazer em javascript assim vc passa na disciplina e estuda javascript

**Comment by ninetails** [comment]
Pense no seguinte: você está fazendo faculdade de Análise de Sistemas, isso vai te formar um analista de sistemas, não um front-end. Mas não pense que escolheu errado, e sim que nossa carreira é muito abrangente e web é só uma parte dela.

Por que fazer essas matérias então? Vou tentar explicar: fazer web não é pegar um job e fazer, isso pode até ser o seu futuro trabalho, mas também falo para não tratar seu trabalho como somente isso. Para se destacar, vc precisa pensar. Faculdade está aí para fazer vc pensar, seja dentro ou fora da caixa. Você vai aprender coisas básicas, como se fosse as matérias que vc aprendeu no ensino médio: Matemática, Filosofia, Português etc. Você não diretamente vai precisar delas, tal como vc provavelmente nunca vai utilizar Cálculo na programação. Mas o que elas te fazem é te dar conhecimento, lhe dar uma base para seu futuro, mesmo que não diretamente utilize. Faculdade ensina vc a estudar.

E não seja preconceituoso com Java. Toda a base de Orientação de Objetos utilizada no dia a dia foi baseada nela por ter sido uma linguagem bem adotada no mundo Enterprise, logo muitos conceitos que tem em Java podem ser portados a Javascript, por exemplo. Aprender Java me deu uma ótima base para Orientação a Objetos com PHP e me abriu portas para aprender Design Patterns, sugiro vc fazer o mesmo. Não estou falando para aprender JBoss, Spring, JWT, Ant. Lá vão te ensinar o básico, e é muito útil esse conhecimento para entender UML, casos de uso, classes, heranças, polimorfismo, composição... E diria o mesmo para as outras matérias (sou preconceituoso somente com Pascal, POR QUE PORRAS ENSINAR PASCAL E NÃO C? UAEHUEAHU)

Por tabela, vc vai conhecer compiladores, que é a idéia dos transpiladores atuais de Javascript (tudo bem, essa idéia é muito C++ que Java mas Node é uma máquina virtual rodando em cima de um V8, por exemplo).

E isso eu to falando só de Javascript, mesmo vendo que vc quer se focar em front-end, muita coisa não vai ser ensinada de qualquer jeito: css, web semântica, acessibilidade, coisas que são aprendidas por fora, mas o que a sua faculdade vai te ajudar é a não depender só de curso, ela vai te forçar vc pegar livros pra estudar, e pra conhecimentos já bem estabelecidos, livros são a melhor forma de estudo dado que n fica um professor te apressando pra aprender, vc aprende no seu ritmo... busca outros livros pra aprender, etc. Mas pelo menos para estas outras coisas, sei que vc vai conseguir se virar bem, dado que vc participa da comunidade.

Também não necessariamente uma graduação é sempre a saída, eu desisti de duas. Tenho amigos também na área de web que não possuem. Por outro lado também conheço muitos que possuem até pós. Não é o fim do mundo não fazer, mas fazer sempre te abre mais portas que não fazer, mesmo que não seja pra web. E também te traz contatos, sejam do corpo docente (professores) ou do corpo discente (amigos que vão ou não graduar com vc, c sempre pode ter o contato deles no LinkedIn).

**Comment by JotaPe** [comment]
Eu acredito que hj as faculdades elas não estão ensinando e sim querendo criar operários que fazem a mesma coisa, isso a um tempo atrás funcionava e ninguém ligava, mas principalmente na área de computação em geral onde existe a constate evolução de mercado não se pode dizer qual linguagem e a melhor para se aprender e sim qual teoria melhor se aprender, Java não e a melhor linguagem para se aprender OOP hj em dia e todo mundo que programa em Java sabe que ela não aplica o paradigma da melhor maneira. Eu já discuti isso antes, mas nem toda linguagem e a melhor implementação do paradigma que ela fala que foi criada para seguir. A solução das faculdades nessa questão e ensinar majoritariamente a teoria sem puxar uma linguagem, algorítimos e logica são muito mais importantes do que uma linguagem de programação a se aprender, então a melhor forma de formar pessoas para nossa área e ensinar essa parte teórica e deixar eles mesmo descobrirem a linguagem que mais se encaixa no futuro que eles querem seguir. E claro que esse modelo e teórico e não conheço nenhuma faculdade que faça isso, mas seria mais interessante que colocar o pessoal para aprender uma linguagem para ensinar a parte teórica, já que eu já vi relatos de professor de faculdade que o primeiro ano na nossa área sempre e cheio um monte de gente vai ver e ir atrás, mas mais da metade da turma some no segundo ano vendo que não era aquilo que eles pensavam. E se eu fosse ir para uma faculdade e tivesse nesse sistema atual eu me sinto que estaria entre essa mais da metade que desiste por estar estudando algo que não gosta, e todo mundo sabe que colégios são assim, vc estuda um monte de matéria chata que pode servir para alguma coisa, mas todos nos sabemos que muita dessa matéria ninguém usa e acaba esquecendo (eu vejo isso claramente na minha mãe e pai que veem a matéria que eu estudo e não lembram de nada que eu estou vendo não por ser novo e sim por não lembrarem de ter estudado), mas se eu fosse dizer algo que vc vai usar querendo ou não em toda área da computação e algoritmos e logica, e como se fosse o português e matemática que a gente aprende no colégio, são as partes básicas que a gente usa pelo resto da vida.

**Comment by Leocardoso94** [comment]
Olha cara, Java é uma linguagem que dá pra fazer de tudo e você aprende bastante conceitos e padrões ao utiliza-lá (principalmente orientação a objetos), no fundo a maioria das linguagens tem suas semelhanças nos mais diversos conceitos, portanto algo que você aprender em Java, será reutilizado um dia em alguma outra linguagem que você estiver utilizando.

**Comment by maxwelbm** [comment]
Mano Java é maravilhoso para aprender orientação a objetos vc aprendendo bem isso Javascript ES6 em diante vai ser molezinha
